javascript/node/selenium-webdriver/lib/webdriver.js

```diff
@@ -1698,6 +1698,23 @@ class TargetLocator {
             setParameter('handle', nameOrHandle));
   }
 
+  /**
+   * Creates a new browser window and switches the focus for future
+   * commands of this driver to the new window.
+   *
+   * @param {string} typeHint 'window' or 'tab'. The created window is not
+   *     guaranteed to be of the requested type; if the driver does not support
+   *     the requested type, a new browser window will be created of whatever type
+   *     the driver does support.
+   * @return {!Promise<void>} A promise that will be resolved
+   *     when the driver has changed focus to the new window.
+   */
+  newWindow(typeHint) {
+    return this.driver_.execute(
```

Comment:
If we follow the same approach as implemented in C# and Java, we should wait until the new window appears and swicth into it.  So it should be something like this:
```
    var driver = this.driver_;
    return this.driver_.execute(
        new command.Command(command.Name.SWITCH_TO_NEW_WINDOW).
            setParameter('type', typeHint)
      ).then(function(response) {
          return driver.switchTo().window(response.handle);
      });
```

Reply:
Just FYI... per the spec the `New Window` command is synchronous.